[AutomaticAssessmentConfiguration] Update Binding to Xcode 12 beta 5

tj-devel709 opened the pull request.

spouliot commented on src/automaticassessmentconfiguration.cs:
```diff
@@ -21,13 +22,76 @@ namespace AutomaticAssessmentConfiguration {
 	[BaseType (typeof (NSObject))]
 	interface AEAssessmentConfiguration : NSCopying {
 
+		[Unavailable (PlatformName.MacCatalyst)]
```
not needed, it's already defined on the type itself

spouliot commented on src/automaticassessmentconfiguration.cs:
```diff
@@ -21,13 +22,76 @@ namespace AutomaticAssessmentConfiguration {
+		[Export ("autocorrectMode")]
+		AEAutocorrectMode AutocorrectMode { get; set; }
+
+		[Unavailable (PlatformName.MacCatalyst)]
```
same

spouliot commented on src/automaticassessmentconfiguration.cs:
```diff
@@ -21,13 +22,76 @@ namespace AutomaticAssessmentConfiguration {
+		[Export ("allowsSpellCheck")]
+		bool AllowsSpellCheck { get; set; }
+
+		[Unavailable (PlatformName.MacCatalyst)]
```
same

spouliot commented on src/automaticassessmentconfiguration.cs:
```diff
@@ -21,13 +22,76 @@ namespace AutomaticAssessmentConfiguration {
+		[Export ("allowsPredictiveKeyboard")]
+		bool AllowsPredictiveKeyboard { get; set; }
+
+		[Unavailable (PlatformName.MacCatalyst)]
```
same

spouliot commented on src/automaticassessmentconfiguration.cs:
```diff
@@ -21,13 +22,76 @@ namespace AutomaticAssessmentConfiguration {
+		[Export ("allowsKeyboardShortcuts")]
+		bool AllowsKeyboardShortcuts { get; set; }
+
+		[Unavailable (PlatformName.MacCatalyst)]
```
same

spouliot commented on src/automaticassessmentconfiguration.cs:
```diff
@@ -21,13 +22,76 @@ namespace AutomaticAssessmentConfiguration {
+		[Export ("allowsActivityContinuation")]
+		bool AllowsActivityContinuation { get; set; }
+
+		[Unavailable (PlatformName.MacCatalyst)]
```
same

spouliot commented on src/automaticassessmentconfiguration.cs:
```diff
@@ -21,13 +22,76 @@ namespace AutomaticAssessmentConfiguration {
+		[Export ("allowsDictation")]
+		bool AllowsDictation { get; set; }
+
+		[Unavailable (PlatformName.MacCatalyst)]
```
same

spouliot commented on src/automaticassessmentconfiguration.cs:
```diff
@@ -21,13 +22,76 @@ namespace AutomaticAssessmentConfiguration {
+		[Export ("allowsAccessibilitySpeech")]
+		bool AllowsAccessibilitySpeech { get; set; }
+
+		[Unavailable (PlatformName.MacCatalyst)]
```
same

spouliot commented on src/automaticassessmentconfiguration.cs:
```diff
@@ -21,13 +22,76 @@ namespace AutomaticAssessmentConfiguration {
+		[Export ("allowsPasswordAutoFill")]
+		bool AllowsPasswordAutoFill { get; set; }
+
+		[Unavailable (PlatformName.MacCatalyst)]
```
same

whitneyschmidt commented on src/automaticassessmentconfiguration.cs:
```diff
@@ -3,6 +3,7 @@
 //	Alex Soto <alexsoto@microsoft.com>
+//	TJ Lambert <TJ.Lambert@microsoft.com>
 //
 // Copyright (c) Microsoft Corporation.
```
```suggestion
// Copyright 2020, Microsoft Corporation.
```

tj-devel709 commented:
I fixed the suggestions and when I tried to push the changes, I got a git error about the tip of my current branch being behind it's remote counterpart and it suggesting a git pull. I had already fetched xamarin and rebased before but did the suggested git pull and when I pushed, tests/xharness/Jenkins/Jenkins.cs had some changes. Is this because someone else has made those changes in the last few hours?

whitneyschmidt commented:
> I fixed the suggestions and when I tried to push the changes, I got a git error about the tip of my current branch being behind it's remote counterpart and it suggesting a git pull. I had already fetched xamarin and rebased before but did the suggested git pull and when I pushed, tests/xharness/Jenkins/Jenkins.cs had some changes. Is this because someone else has made those changes in the last few hours?

You can check the commits on xcode12 to see whether anything new was committed :) It looks like UIKit changes were merged 7 min ago.

dalexsoto commented on src/automaticassessmentconfiguration.cs:
```diff
@@ -3,8 +3,9 @@
+//	TJ Lambert <TJ.Lambert@microsoft.com>
 //
-// Copyright (c) Microsoft Corporation.
+// Copyright (2020) Microsoft Corporation.
```
Do not change copyright headers, this is correct as it was

```suggestion
// Copyright (c) Microsoft Corporation.
```

dalexsoto commented on src/automaticassessmentconfiguration.cs:
```diff
@@ -21,13 +22,67 @@ namespace AutomaticAssessmentConfiguration {
 	interface AEAssessmentConfiguration : NSCopying {
 
+		[NoMac, iOS (14, 0)]
+		[Advice ("This API is not available when using UIKit on macOS.")]
```
This is not needed as the parent class already holds this advice

```suggestion
		[Advice ("This API is not available when using UIKit on macOS.")]
```

dalexsoto replied:
same for all the ones below within `AEAssessmentConfiguration`

tj-devel709 commented:
My branch ended up getting polluted so with Alex's help, I am closing this PR and creating a new PR with a clean branch! :) Sorry for any inconvenience!
